ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ValueNetsingIT.scala

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites
+
+import com.daml.ledger.api.refinements.ApiTypes.Party
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.test.semantic.ValueNesting._
+import io.grpc.Status
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+final class ValueNestingIT extends LedgerTestSuite {
+
+  private[this] def toEither[X](future: Future[X])(implicit
+      ec: ExecutionContext
+  ): Future[Either[Throwable, X]] =
+    future.transform(x => Success(x.toEither))
+
+  List[Long](30, 100, 101, 110, 200).foreach { depth =>
+    val accepted = depth <= 100
+    val result = if (accepted) "Accept" else "Reject"
+
+    // Once converted to Nat, `depthNat` will have depth `depth`.
+    // Note that Nat.Z has depth 2.
```

Comment:
Outer Value wrapper + variant in the oneof or how do we get to 2?

Reply:
Variant constructors have exactly one argument in LF. In this case the compiler field the argument of `Z` with a Unit.